lib/sqlitex/statement.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Sqlitex.Statement do
+  @moduledoc """
+  Provides an interface for working with sqlite prepared statements.
+
+  ```
+  iex(2)> {:ok, db} = Sqlitex.open(":memory:")
+  iex(3)> Sqlitex.query(db, "CREATE TABLE data (id, name);")
+  []
+  iex(6)> {:ok, statement} = Sqlitex.Statement.prepare(db, "INSERT INTO data VALUES (?, ?);")
+  iex(7)> Sqlitex.Statement.bind_values(statement, [1, "hello"])
+  iex(8)> Sqlitex.Statement.fetch_all(statement)
```

Comment:
I wonder if we should have a `Sqlitex.Statement.exec` function. It seems odd to be to call a function named `fetch_all` when you know that it doesn't actually have any data to be collected/returned.

I think for the same reasons that we have `Sqlitex.query` and `Sqlitex.exec` we should probably have a `Sqlitex.Statement.exec` which just executes a prepared statement and skips all the data collection steps.